standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -546,9 +548,17 @@ public static void updateTableStatsForCreateTable(Warehouse wh, Database db, Tab
       }
     }
 
-    if (MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.STATS_AUTO_GATHER)) {
+    boolean isIcebergTable = Optional.ofNullable(tbl.getParameters())
+            .orElse(Collections.emptyMap())
+            .getOrDefault(TABLE_TYPE_PROP, "undefined")
+            .equals(ICEBERG_TABLE_TYPE_VALUE);
+
+    if (MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.STATS_AUTO_GATHER) && !isIcebergTable) {
```

Comment:
Is it a good idea to hardcode such specific case in the metastore server? Why not directly disable `STATS_AUTO_GATHER`?

Reply:
This won't work for the following scenario: let's suppose stats source is Iceberg and you create an Iceberg table and add a parameter to the table 'DO_NOT_UPDATE_STATS=true'. Then you change stats source to Metastore. Stats won't be gathered for this table because of the table's parameter.